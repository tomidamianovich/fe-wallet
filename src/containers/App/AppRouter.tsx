import { Suspense } from 'react';
import { BrowserRouter, Route, Routes, Navigate } from 'react-router-dom';
import Navigation from '../../components/Navigation';
import Header from '../../components/Header';
import {
  Panel,
  Wallet,
  Credit,
  Exchange,
  Launchpad,
  MarketRates,
  NotFound
} from '../../containers';
import { NavLinkType, CombinedState } from '../../utils/type'
import { useSelector } from 'react-redux';

export default function AppRouter() {
  const navLinks:NavLinkType  = useSelector((state:CombinedState) => state.NavLinkReducer);
  
  const LoadingMessage = () => <div>Loading...</div>;

  return (
    <BrowserRouter>
      <Header />
      <div className='fe-wallet__content'>
        <Suspense fallback={<LoadingMessage />}>
          <Navigation links={navLinks} />
          <Routes>
            <Route path="/panel"  element={<Panel />} />
            <Route path="/wallet" element={<Wallet />} />
            <Route path="/market-rates" element={<MarketRates />} />
            <Route path="/credit" element={<Credit />} />
            <Route path="/exchange" element={<Exchange />} />
            <Route path="/launchpad" element={<Launchpad />} />
            <Route path="*" element={<NotFound />} />
            <Route path="/" element={<Navigate to="/panel" />}/>  
          </Routes>
        </Suspense>
      </div>
    </BrowserRouter>
  );
}
