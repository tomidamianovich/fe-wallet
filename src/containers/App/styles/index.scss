@import "../styles/colours.scss";

html, body, #root {
  width: 100%;
  height: 100%;
  font-family: 'Lato';
}

.fe-wallet {
  width: 100%;
  height: 100%;
  font-family: 'Lato';
  text-align: center;
  background-color: $primary-background-color;

  &__content {
    display: flex;
    height: calc( 100% - 80px);
  }
}