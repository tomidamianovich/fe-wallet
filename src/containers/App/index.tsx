import AppRouter  from './AppRouter';
import './styles/index.scss';

function App() {
  return (
    <div className="fe-wallet">
      <AppRouter />
    </div>
  );
}

export default App;
