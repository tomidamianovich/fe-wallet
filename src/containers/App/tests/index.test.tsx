import { render } from '@testing-library/react';
import App from '../';
import { createStore, applyMiddleware } from "redux";
import { Provider } from 'react-redux';

const store = createStore(()=>[], applyMiddleware());

jest.mock('react-i18next', () => ({
  withTranslation: () => (Component: any) => {
    Component.defaultProps = {...Component.defaultProps, t: (translation: string) => translation};
    return Component;
  }
}));

test('renders App main page', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <App />
    </Provider>);
  expect(asFragment()).toMatchSnapshot();
});
