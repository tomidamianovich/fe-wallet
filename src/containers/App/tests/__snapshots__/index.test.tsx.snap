// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App main page 1`] = `
<DocumentFragment>
  <div
    class="fe-wallet"
  >
    <header
      class="fe-wallet__header"
    >
      <div
        class="fe-wallet__header__logo-container"
      >
        <a
          aria-current="page"
          class="active"
          href="/panel"
        >
          <img
            alt="logo"
            class="fe-wallet__header__logo-container__logo"
            src="logo-brand.svg"
          />
        </a>
        <div
          class="fe-wallet__header__icons"
        >
          <svg
            class="icon"
            height="26"
            viewBox="0 0 26 26"
            width="26"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M13 24.2C19.1856 24.2 24.2 19.1856 24.2 13C24.2 6.81442 19.1856 1.80001 13 1.80001C6.81442 1.80001 1.80001 6.81442 1.80001 13C1.80001 19.1856 6.81442 24.2 13 24.2ZM13 25.8C5.93077 25.8 0.200012 20.0693 0.200012 13C0.200012 5.93077 5.93077 0.200012 13 0.200012C20.0693 0.200012 25.8 5.93077 25.8 13C25.8 20.0693 20.0693 25.8 13 25.8ZM9.80001 7.87481C9.99658 7.70137 10.2119 7.5366 10.4461 7.3805C10.6802 7.2244 10.9332 7.08854 11.2049 6.97291C11.4766 6.85728 11.7715 6.76623 12.0894 6.69974C12.4074 6.63325 12.7514 6.60001 13.1214 6.60001C13.6244 6.60001 14.0826 6.66939 14.4959 6.80814C14.9093 6.9469 15.2634 7.14491 15.5583 7.40218C15.8531 7.65945 16.0815 7.9702 16.2434 8.33443C16.4053 8.69866 16.4862 9.10624 16.4862 9.55719C16.4862 9.99658 16.4226 10.3767 16.2954 10.6976C16.1682 11.0184 16.0078 11.2974 15.8141 11.5344C15.6204 11.7715 15.4109 11.9767 15.1854 12.1501C14.9599 12.3236 14.746 12.4869 14.5436 12.6401C14.3413 12.7933 14.1664 12.9436 14.019 13.0911C13.8716 13.2385 13.7834 13.4047 13.7545 13.5897L13.555 14.8558H12.0894L11.942 13.4423C11.9362 13.4134 11.9333 13.3888 11.9333 13.3686V13.2949C11.9333 13.0405 11.9969 12.8193 12.1241 12.6314C12.2513 12.4436 12.4103 12.2672 12.6011 12.1024C12.7919 11.9377 12.9971 11.7772 13.2168 11.6211C13.4365 11.4651 13.6417 11.2959 13.8325 11.1138C14.0233 10.9317 14.1823 10.725 14.3095 10.4938C14.4367 10.2625 14.5003 9.9908 14.5003 9.6786C14.5003 9.47047 14.4613 9.28402 14.3832 9.11925C14.3052 8.95448 14.1968 8.81284 14.058 8.69432C13.9193 8.5758 13.753 8.48475 13.5594 8.42115C13.3657 8.35755 13.1561 8.32576 12.9306 8.32576C12.6011 8.32576 12.3221 8.36189 12.0938 8.43416C11.8654 8.50643 11.6717 8.58736 11.5127 8.67698C11.3538 8.76659 11.2193 8.84753 11.1095 8.91979C10.9996 8.99206 10.9014 9.0282 10.8146 9.0282C10.6065 9.0282 10.4562 8.94148 10.3637 8.76803L9.80001 7.87481ZM11.439 18.0992C11.439 17.92 11.4723 17.7494 11.5388 17.5875C11.6053 17.4257 11.6978 17.2869 11.8163 17.1713C11.9348 17.0557 12.075 16.9632 12.2369 16.8938C12.3987 16.8244 12.5722 16.7897 12.7572 16.7897C12.9364 16.7897 13.107 16.8244 13.2688 16.8938C13.4307 16.9632 13.5695 17.0557 13.6851 17.1713C13.8007 17.2869 13.8932 17.4257 13.9626 17.5875C14.032 17.7494 14.0667 17.92 14.0667 18.0992C14.0667 18.2842 14.032 18.4562 13.9626 18.6152C13.8932 18.7742 13.8007 18.9115 13.6851 19.0271C13.5695 19.1427 13.4307 19.2338 13.2688 19.3003C13.107 19.3668 12.9364 19.4 12.7572 19.4C12.5722 19.4 12.3987 19.3668 12.2369 19.3003C12.075 19.2338 11.9348 19.1427 11.8163 19.0271C11.6978 18.9115 11.6053 18.7742 11.5388 18.6152C11.4723 18.4562 11.439 18.2842 11.439 18.0992Z"
              fill-rule="evenodd"
            />
          </svg>
          <svg
            class="icon"
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M15.6314 4.80001C15.1724 4.80001 14.7299 5.20463 14.7299 5.79294V8.6631L14.1021 8.80149C10.7365 9.54346 8.16024 12.7082 8.16024 16.553V16.5969L7.47757 22.8079L7.09162 23.0037C6.69426 23.2053 6.39999 23.6421 6.39999 24.1728V24.3721C6.39999 24.5615 6.43638 24.6317 6.44456 24.6456L6.44503 24.6464C6.44874 24.6528 6.4584 24.6696 6.51315 24.6945C6.58316 24.7264 6.70219 24.7591 6.89731 24.7798C7.09015 24.8002 7.31567 24.8053 7.58617 24.8053H23.6748C23.9452 24.8053 24.1706 24.8002 24.3635 24.7798C24.5587 24.7591 24.6779 24.7264 24.7481 24.6944C24.8032 24.6694 24.813 24.6524 24.8168 24.6459L24.8172 24.6451C24.8255 24.6312 24.8619 24.5609 24.862 24.3717V24.1729C24.862 23.6427 24.5684 23.2068 24.1703 23.0045L23.7847 22.8086L23.1021 16.597V16.5532C23.1021 12.7079 20.5258 9.54348 17.1602 8.80166L16.5324 8.66328V5.79294C16.5324 5.20449 16.0898 4.80001 15.6314 4.80001ZM13.1299 5.79294C13.1299 4.40105 14.2108 3.20001 15.6314 3.20001C17.0517 3.20001 18.1324 4.4012 18.1324 5.79294V7.40232C21.9349 8.54355 24.6831 12.2055 24.702 16.5085L25.2856 21.8188C26.0011 22.3452 26.462 23.2054 26.462 24.1729V24.3722C26.4618 24.7614 26.3867 25.1351 26.1937 25.4608C25.9959 25.7946 25.713 26.0132 25.4108 26.1507C24.8669 26.3983 24.1909 26.4053 23.6748 26.4053H19.6105C19.4205 27.0314 19.0047 27.5613 18.4845 27.9514C17.7658 28.4902 16.8098 28.7997 15.7881 28.7997C14.7664 28.7997 13.8103 28.4902 13.0917 27.9514C12.5713 27.5613 12.1554 27.0314 11.9654 26.4053H7.58617C7.06995 26.4053 6.39383 26.3983 5.84997 26.1506C5.54784 26.013 5.26501 25.7941 5.06746 25.4602C4.8747 25.1343 4.79999 24.7612 4.79999 24.3721V24.1728C4.79999 23.2057 5.2608 22.3443 5.97678 21.8177L6.56034 16.5083C6.57926 12.2058 9.32747 8.54357 13.1299 7.40219V5.79294ZM13.8016 26.4487C13.873 26.5252 13.9561 26.5998 14.0514 26.6713C14.4675 26.9832 15.0789 27.1997 15.7881 27.1997C16.4973 27.1997 17.1086 26.9832 17.5246 26.6713C17.62 26.5998 17.703 26.5253 17.7744 26.4487H13.8016Z"
              fill-rule="evenodd"
            />
          </svg>
          <svg
            class="icon"
            height="29"
            viewBox="0 0 32 29"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.53858 28.1065C2.14635 25.1726 0 20.837 0 16C0 7.16344 7.16344 0 16 0C24.8366 0 32 7.16344 32 16C32 20.837 29.8537 25.1726 26.4614 28.1065C25.2648 23.4451 21.0347 20 16 20C10.9653 20 6.73524 23.4451 5.53858 28.1065ZM16 18.4C19.0928 18.4 21.6 15.8928 21.6 12.8C21.6 9.7072 19.0928 7.2 16 7.2C12.9072 7.2 10.4 9.7072 10.4 12.8C10.4 15.8928 12.9072 18.4 16 18.4Z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </header>
    <div
      class="fe-wallet__content"
    >
      <nav
        class="fe-wallet__content__navigation"
      >
        <div
          class="fe-wallet__content__navigation__logout"
        >
          <li
            class="fe-wallet__content__navigation__list__item"
          >
            <a
              class="fe-wallet__content__navigation__list__item__link false"
              href="/logout"
            >
              <div
                class="fe-wallet__content__navigation__list__item__link__content"
              >
                <svg
                  class="icon"
                  height="16"
                  viewBox="0 0 15 16"
                  width="15"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M2.39378 14.1768H7.9752C8.31814 14.1768 8.59534 14.454 8.59534 14.7969C8.59534 15.1398 8.31814 15.417 7.9752 15.417H2.39378C1.36805 15.417 0.533325 14.5823 0.533325 13.5566V2.39381C0.533325 1.36808 1.36805 0.533325 2.39378 0.533325H7.9752C8.31814 0.533325 8.59534 0.810584 8.59534 1.15353C8.59534 1.49647 8.31814 1.77367 7.9752 1.77367H2.39378C2.05148 1.77367 1.77364 2.05148 1.77364 2.39381V13.5566C1.77364 13.8989 2.05145 14.1768 2.39378 14.1768ZM11.5113 3.81271L15.2819 7.53365C15.4003 7.65087 15.4667 7.80901 15.4667 7.97514C15.4667 8.14133 15.3997 8.30012 15.2819 8.4167L11.5113 12.1376C11.391 12.2567 11.2335 12.3162 11.076 12.3162C10.916 12.3162 10.7559 12.2542 10.6344 12.1314C10.3944 11.8877 10.3963 11.4952 10.6406 11.2546L13.3352 8.59534H6.11472C5.77177 8.59534 5.49458 8.31811 5.49458 7.97517C5.49458 7.63223 5.77177 7.35503 6.11472 7.35503H13.3352L10.6406 4.69582C10.3963 4.45519 10.3938 4.06264 10.6344 3.81892C10.875 3.5752 11.2682 3.57208 11.5113 3.81271Z"
                    fill-rule="evenodd"
                  />
                </svg>
                <span>
                  Salir
                </span>
              </div>
            </a>
          </li>
        </div>
      </nav>
      <div
        class="fe-wallet__panel"
      >
        <main
          class="fe-wallet__main"
        >
          <div
            class="fe-wallet__spinner-container"
          >
            <div
              class="fe-wallet__spinner-container__content"
            />
          </div>
          <div
            class="fe-wallet__spinner-container"
          >
            <div
              class="fe-wallet__spinner-container__content"
            />
          </div>
          <div
            class="fe-wallet__spinner-container"
          >
            <div
              class="fe-wallet__spinner-container__content"
            />
          </div>
          <div
            class="fe-wallet__transactions"
          >
            <span>
              Mis movimientos
            </span>
            <div
              class="fe-wallet__transactions__container"
            >
              <table>
                <thead>
                  <tr>
                    <th>
                      Fecha
                    </th>
                    <th>
                      Tipo de Operación
                    </th>
                    <th>
                      Método
                    </th>
                    <th>
                      Estado
                    </th>
                    <th>
                      Comisión
                    </th>
                    <th>
                      Monto
                    </th>
                  </tr>
                </thead>
                <tbody />
              </table>
            </div>
          </div>
        </main>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
